_posts/2024-02-21-securing-wildfly-apps-okta-openshift.adoc

```diff
@@ -0,0 +1,262 @@
+---
+layout: post
+title: 'Securing WildFly Applications Using Okta on OpenShift'
+date: 2024-02-21
+tags: oidc OpenShift okta
+synopsis: A step by step guide on how to secure a WildFly application using Okta's OpenID Provider and deploy it to OpenShift. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+WildFly applications can be secured using OpenID Connect and deployed to OpenShift. By using OIDC to secure applications, you delegate authentication to OIDC providers. The `elytron-oidc-client` subsystem can be used to secure an application deployed to WildFly using any OpenID Provider. This guide demonstrates how to secure an example application deployed to WildFly on OpenShift using https://www.okta.com/[Okta] as the OpenID Provider. 
+
+toc::[] 
+
+== Prerequisites 
+To follow along this guide you will need: 
+
+* Roughly 15 minutes
+* Access to an OpenShift cluster (try the https://developers.redhat.com/developer-sandbox[Red Hat Developer Sandbox] for free)
+* https://docs.openshift.com/container-platform/4.14/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* https://helm.sh/docs/intro/install/[Helm Chart]
+* https://www.okta.com/[Okta OpenID Provider]
+
+== Example application 
+
+We will use a simple web application in this guide that consists of a single https://github.com/wildfly-security-incubator/elytron-examples/blob/simple-webapp-okta/simple-webapp-okta/src/main/java/org/wildfly/security/examples/SecuredServlet.java[servlet]. We will use the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/simple-webapp-okta[simple-webapp-okta] application in this repository.
```

Comment:
The servlet link doesn't work, I think it should be:

https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-okta/src/main/java/org/wildfly/security/examples/SecuredServlet.java